### ISSUE TITLE
【Question】Reproduce Appendix F (SEEDV)

### ISSUE THREAD

**Issue by ShogoNoguchi** [935963004/LaBraM#52]
Hello, thank you for your great work.

Our research team is interested in emotion classification, so we have forked your work and created code for SEEDV to reproduce Appendix F (SEED-V) in your paper.

Specifically, we created make_SEED.py (see make_SEED.ipynb for the "output") to preprocess SEEDV, added a branch for SEEDV to the get_dataset function in run_class_finetuning.py, and added the SEEDVLaoder class and prepare_SEEDV_dataset function to utils.py. (We will make a pull request when the results are available.)

We are currently in the process of learning, but would like to ask the following:
1. Will my make_SEED.py be able to obtain the preprocessing described in the paper?
2. Currently, we are getting messages such as "NaN or Inf found in input tensor." and "grad_norm: 1.5252 (inf)" in various places. Is this okay?

**Comment by ShogoNoguchi** [comment]
Also, I am using all 62 channels, is this correct for reproducing the paper?I thought it would be okay because there is absolute spatial position embedding...

**Comment by 935963004** [comment]
Thanks for your effort! It is a little bit different between the SEED-V (20 subjects) we used in the paper and the public SEED-V (16 subjects). Our used one is a private one with 4 more subjects, which may lead to minor differences, and this is why we did not publish the code for SEED-V. However, I think the results should be consistent.
Your make_SEEDV.py seems correct, but I noticed that you seem to encounter failed files which I did not have in my experiment. Using all 62 channels is correct and you can ignore the NaN warning if it does not affect the training. Did you ensure all hyperparameters for finetuning are consistent with the paper?

**Comment by ShogoNoguchi** [comment]
Thank you for taking the time to reply despite your busy schedule! Your reply is very, very helpful to me! (Sorry for the long question.)

I see, I understand the situation regarding the SEEDV paper. Thank you very much for checking make_SEEDV.py!

I was a little bothered by inconsistencies in the annotations (data acquisition date and time, which does not affect training) when processing SEEDV, but I dealt with it by adjusting make_SEEDV.py, and now all data for 16 people is usable!

I started training with the following execution code. I believe this is in accordance with Appendix C of the paper! I would appreciate it if you could check it.

※All the execution code is in this ipynb! (My execution environment is colab).
[https://github.com/ShogoNoguchi/fork-LaBraM_Task-CSL/blob/main/Finetuning%20execution%20code%20in%20colab.ipynb]
This execution code isn't long, so I hope you'll take a look!

OMP_NUM_THREADS=1 python run_class_finetuning.py \ 
--output_dir ./checkpoints/finetune_tuab_base/ \ 
--log_dir ./log/finetune_tuab_base \
 --model labram_base_patch200_200 \
 --finetune ./checkpoints/labram-base.pth \ 
--weight_decay 0.05 \
 --batch_size 512 \ 
--lr 5e-4 \ 
--update_freq 1 \
 --warmup_epochs 5 \
 --epochs 50 \
 --layer_decay 0.65 \
 --drop_path 0.1 \ 
--dist_eval \ 
--save_ckpt_freq 5 \
 --disable_rel_pos_bias \ 
--abs_pos_emb \
 --dataset SEEDV \ 
--disable_qkv_bias \
--seed 0 \
--resume ./checkpoints/finetune_tuab_base/checkpoint.pth

※--resume is only used when starting from the middle.

※I removed the # from utils.auto_load_model()

args.start_epoch = 1#checkpoint['epoch'] + 1 and set it to args.start_epoch = checkpoint['epoch'] + 1
to allow the training to resume even from the middle.

Let me ask a new question:
I prevented the following warning by shuffling the evaluation set and test set once in the prepare_SEEDV function (my own creation) in utils.py. Is this the correct way to deal with it?
warnings.warn("y_pred contains classes not in y_true")

prepare_SEEDV function:
ーーーーーーーー
def prepare_SEEDV_dataset(root):
import random
seed = 12345
np.random.seed(seed)
random.seed(seed)

"processed_train"))
val_files = os.listdir(os.path.join(root, "processed_eval"))
test_files = os.listdir(os.path.join(root, "processed_test"))

# Shuffle validation and test files only the first time (one time only)
random.shuffle(val_files)
random.shuffle(test_files)

train_dataset = SEEDVLoader(
os.path.join(root, "processed_train"), train_files
)
val_dataset = SEEDVLoader(
os.path.join(root, "processed_eval"), val_files
)
test_dataset = SEEDVLoader(
os.path.join(root, "processed_test"), test_files
)

print("SEEDV dataset stats:")
print("train:", len(train_files), "val:", len(val_files), "test:", len(test_files))
return train_dataset, test_dataset, val_dataset
ーーーーーー



Thank you in advance!


*All execution codes are available at [https://github.com/ShogoNoguchi/fork-LaBraM_Task-CSL/tree/main], so please check them out as necessary.

**Comment by 935963004** [comment]
I think your hyperparameters are okay and the shuffling does not affect the results. Can you show me the tensorboard visualization of the whole training procedure for the training, val, and test metrics?

**Comment by ShogoNoguchi** [comment]
Thank you for your reply!!!

OK, this is my log up to the 5th epoch.
#I'm still in the early stages of learning, but I hope you'll take a look!


![スクリーンショット 2024-12-13 134525](https://github.com/user-attachments/assets/2d9f3d70-1bd6-436b-aed9-7cceb27f43fb)
![スクリーンショット 2024-12-13 134533](https://github.com/user-attachments/assets/4223fd2d-c78d-408b-ad8e-ba9575114000)
![スクリーンショット 2024-12-13 134541](https://github.com/user-attachments/assets/572e2a63-7fe6-40f0-a56e-a59f7e7de200)
![スクリーンショット 2024-12-13 134550](https://github.com/user-attachments/assets/262f4db7-289c-484b-a1f6-136b406440af)
![スクリーンショット 2024-12-13 134556](https://github.com/user-attachments/assets/76e205ce-453c-421f-a294-7b27085f0e7e)
![スクリーンショット 2024-12-13 134609](https://github.com/user-attachments/assets/f6d20826-76b5-49e7-bed7-40044fd24d22)
![スクリーンショット 2024-12-13 134615](https://github.com/user-attachments/assets/240f0508-3701-4629-8713-1ada65cc963a)

**Comment by 935963004** [comment]
It is a little strange. In my experiment, I remember that the value of all metrics consistently increases as the training procedure goes and gradually converges to a certain value.

**Comment by ShogoNoguchi** [comment]
Thank you for your reply!!

That's a bit scary...
I'll show you again when the log is updated, so I hope you'll take a look!

**Comment by 935963004** [comment]
In my case, I use **raw = mne.io.read_raw_cnt(cntFilePath, preload=True, data_format='int32')** to read the cnt files, which is a little different from yours. We did not specify the data_formate and data_format is int32. Maybe you can try this?

**Comment by ShogoNoguchi** [comment]
Thanks for the reply!!!
Thank you so much for sharing that important information! I'll give it a try!

**Comment by 935963004** [comment]
I think it would be more convenient for us to communicate using social media such as line (maybe I can register one if you want). You can send me your information through my email in my paper if you want. I would be happy to help you replicate the results on SEED-V.

**Comment by ShogoNoguchi** [comment]
That's a very exciting proposal!
Please let me do so, I will do my best to replicate the SEEDV results in this paper!!
Please wait a moment. I'll do it right away.
